consensus/service/src/peer_api_service.rs

```diff
@@ -363,3 +377,318 @@ impl<E: ConsensusEnclave, L: Ledger, TXM: TxManager + Clone> ConsensusPeerApi
+    };
+    use grpcio::{ChannelBuilder, Environment, Error::RpcFailure, Server, ServerBuilder};
+    use mc_common::{
+        logger::{test_with_logger, Logger},
+        NodeID, ResponderId,
+    };
+    use mc_consensus_api::{
+        consensus_peer::{ConsensusMsg, ConsensusMsgResult},
+        consensus_peer_grpc,
+        consensus_peer_grpc::ConsensusPeerApiClient,
+    };
+    use mc_consensus_enclave_mock::MockConsensusEnclave;
+    use mc_consensus_scp::{
+        msg::{NominatePayload, Topic::Nominate},
+        Msg, QuorumSet,
+    };
+    use mc_crypto_keys::{Ed25519Pair, Ed25519Private};
+    use mc_ledger_db::MockLedger;
+    use mc_peers;
+    use mc_transaction_core::{tx::TxHash, Block};
+    use mc_util_from_random::FromRandom;
+    use rand::{rngs::StdRng, SeedableRng};
+    use std::sync::Arc;
+
+    // Get sensibly-initialized mocks.
+    fn get_mocks() -> (MockConsensusEnclave, MockLedger, MockTxManager) {
+        let consensus_enclave = MockConsensusEnclave::new();
+        let ledger = MockLedger::new();
+        let tx_manager = MockTxManager::new();
+
+        (consensus_enclave, ledger, tx_manager)
+    }
+
+    /// Always returns OK.
+    fn get_incoming_consensus_msgs_sender_ok(
+    ) -> Arc<dyn Fn(IncomingConsensusMsg) -> Result<(), BackgroundWorkQueueError> + Sync + Send>
+    {
+        Arc::new(|_msg: IncomingConsensusMsg| {
+            // TODO: store inputs for inspection.
+            Ok(())
+        })
+    }
+
+    // Does nothing.
+    fn get_scp_client_value_sender(
+    ) -> Arc<dyn Fn(TxHash, Option<&NodeID>, Option<&ResponderId>) + Sync + Send> {
+        Arc::new(
+            |_tx_hash: TxHash, _node_id: Option<&NodeID>, _responder_id: Option<&ResponderId>| {
+                // Do nothing.
+            },
+        )
+    }
+
+    // Returns None.
+    fn get_fetch_latest_msg_fn() -> Arc<dyn Fn() -> Option<mc_peers::ConsensusMsg> + Sync + Send> {
+        Arc::new(|| None)
+    }
+
+    fn get_client_server(instance: PeerApiService) -> (ConsensusPeerApiClient, Server) {
+        let service = consensus_peer_grpc::create_consensus_peer_api(instance);
+        let env = Arc::new(Environment::new(1));
+        let mut server = ServerBuilder::new(env.clone())
+            .register_service(service)
+            .bind("127.0.0.1", 0)
+            .build()
+            .unwrap();
+        server.start();
+        let (_, port) = server.bind_addrs().next().unwrap();
+        let ch = ChannelBuilder::new(env).connect(&format!("127.0.0.1:{}", port));
+        let client = ConsensusPeerApiClient::new(ch);
+        (client, server)
+    }
+
+    #[test_with_logger]
+    // Should ignore a message from an unknown peer.
+    fn test_send_consensus_msg_ignore_unknown_peer(logger: Logger) {
+        let (consensus_enclave, ledger, tx_manager) = get_mocks();
+
+        // ResponderIds seem to be "host:port" strings.
+        let known_responder_ids = vec![
+            ResponderId("A:port".to_string()),
```

Comment:
same nit on all these string literals; my understanding is that `to_owned` is preferred